src/vireshJJun20/assignment33/StringOpsItemId.java

```diff
@@ -0,0 +1,37 @@
+package vireshJJun20.assignment33;
+
+public class StringOpsItemId {
+
+	void getSubString(String ipStr) {
+		String tempStr1 = "", tempStr2 = "";
+		for (int index = 0; index < ipStr.length(); index++) {
+			if (ipStr.charAt(index) == '=' && ipStr.charAt(index - 3) == 'm') {
+				for (int innerIndex = index - 6; innerIndex < ipStr.length(); innerIndex++) {
+					if (innerIndex < index) {
+						tempStr1 += String.valueOf(ipStr.charAt(innerIndex));
+					} else if (innerIndex > index && Character.isDigit(ipStr.charAt(innerIndex)) == true
+							|| innerIndex == index) {
+						tempStr2 += String.valueOf(ipStr.charAt(innerIndex));
+					} else
+						break;
+				}
+			}
+		}
+		if (tempStr1.isEmpty() || !(tempStr1.equalsIgnoreCase("itemid"))) {
+			System.out.println(ipStr);
+			System.out.println("No item id present");
+		} else if (tempStr1.equalsIgnoreCase("itemid")) {
+			System.out.println(ipStr);
+			System.out.println(tempStr1.substring(0, tempStr1.length() - 2) + " "
+					+ tempStr1.substring(tempStr1.length() - 2) + " is " + tempStr2.substring(1, tempStr2.length()));
```

Comment:
Lots of complexity. 

1st way :
if (input.contains("itemId=")) {
			String str = input.split("itemId=")[1];
			String output = "";
			for (int index = 0; index < str.length(); index++) {
				if (Character.isDigit(str.charAt(index))) {
					output += str.charAt(index);
				} else {
					break;
				}
			}
			System.out.println("Item id is " + output);
		} else {
			System.out.println("No itemID present.");
		}

2nd way:
if (input.contains("itemId")) {
			int item = input.indexOf("itemId");
			String sub = input.substring(item);
			String s = sub.split("=")[1].split("&")[0].split("/")[0];
			System.out.println("ItemID is " + s);
		} else {
			System.out.println("No itemID present.");
		}

3rd and optimize way:

try{
String str = input.split("itemId=")[1].replace("/", "").split("&")[0];
System.out.println("Item id is " + str);

}catch(StringIndexOutOfBoundsException siob){
			System.out.println("Item ID -> Not present");
		}